packages/react-kit/src/stories/helpers/connect-wallet.ts

```diff
@@ -1,19 +1,42 @@
 import { initializeConnector } from "@web3-react/core";
 import { MetaMask } from "@web3-react/metamask";
+import { utils } from "ethers";
 
-export const [metaMask, hooks] = initializeConnector<MetaMask>(
-  (actions) => new MetaMask(actions)
-);
+import { Web3ReactHooks } from "@web3-react/core";
 
-export function connectWallet(chainId?: number) {
-  metaMask.provider
-    ?.request({
-      method: "wallet_requestPermissions",
-      params: [
-        {
-          eth_accounts: {}
-        }
-      ]
-    })
-    .then(() => metaMask.activate(chainId));
-}
+export const [metaMask, hooks] = initializeConnector<MetaMask>((actions) => {
+  return new MetaMask({ actions });
+});
+
+export const connectWallet = async (chainId?: string) => {
+  try {
+    if (
+      chainId &&
+      window.ethereum &&
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      window.ethereum.networkVersion !== chainId
+    ) {
+      try {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        await window.ethereum.request({
+          method: "wallet_switchEthereumChain",
+          params: [{ chainId: utils.hexStripZeros(utils.hexlify(chainId)) }]
+        });
+      } catch (err) {
+        console.error("Network changed rejected", err);
+      }
```

Comment:
```suggestion
    const currentChainId = window.ethereum
      ? Number(
          await (window.ethereum as any)?.request({ method: "eth_chainId" })
        )
      : undefined;
    if (
      chainId &&
      window.ethereum &&
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      currentChainId !== Number(chainId)
    ) {
      try {
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        await window.ethereum.request({
          method: "wallet_switchEthereumChain",
          params: [
            { chainId: utils.hexStripZeros(utils.hexlify(Number(chainId))) }
          ]
        });
      } catch (err) {
        console.error("Network changed rejected", err);
      }
```

Reply:
- detection of the current network with `window.ethereum.networkVersion` does not work on my machine (Firefox 124.0.1 / Metamask 11.12.2)
- ethers.utils.hexlify("80001") fails with error` invalid hexlify value (argument="value", value="80001", code=INVALID_ARGUMENT, version=bytes/5.7.0)`. Value is expected to be a number, or a string starting with "0x"